### PR TITLE
Add Log Operator

### PR DESCRIPTION
**Pull request by MARD1NO**
### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### Performance

image

### CODE REVIEW

**Review by StrongSpoon — COMMENTED** [review]
please implement benchmark for log.

**Comment by MARD1NO** [comment]
> please implement benchmark for log.

Added log benchmark

**Review by StrongSpoon — APPROVED** [review]
lgtm

File: tests/test_unary_pointwise_ops.py [hunk 1]
```diff
@@ -780,6 +780,20 @@ def test_accuracy_logical_not(shape, dtype):
     gems_assert_equal(res_out, ref_out)
 
 
+@pytest.mark.log
+@pytest.mark.parametrize("shape", POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_accuracy_log(shape, dtype):
+    inp = torch.rand(shape, dtype=dtype, device=flag_gems.device)
+
+    ref_inp = to_reference(inp)
```
Comment (StrongSpoon) [inline tests/test_unary_pointwise_ops.py:789]: but I suggest upcasting reference input here to get higher accuracy.

File: tests/test_unary_pointwise_ops.py [hunk 2]
```diff
@@ -780,6 +780,20 @@ def test_accuracy_logical_not(shape, dtype):
...
+    ref_out = torch.log(ref_inp)
+    with flag_gems.use_gems():
+        res_out = torch.log(inp)
+
+    gems_assert_equal(res_out, ref_out)
```
Comment (StrongSpoon) [inline tests/test_unary_pointwise_ops.py:794]: as for vendors need to keep aligned with cpu reference, I suggest upcasting the reference input to fp64 and asserting close for outputs.
